### ISSUE TITLE
Authentication - Internal Credentials

### ISSUE THREAD

**Issue by wreality** [MESH-Research/CCR#63]
**Problem/Why**

There is currently no way to identify users in the system. There is also no way for users to authenticate using their email address and a custom password.

---

**Goal/What**

Create a system that allows users to authenticate. Its sign in functionality should accept email and custom password pairs. Verify that the email exists in the system and that the password typed in matches that email.

---

**Solution/How**

Create data structures in GraphQL to represent users.

Write backend code to create necessary components/models for representing a user and its credentials.

Write front-end code to send credentials to backend.

Write front-end code to handle the response from the backend. 
### Acceptance Criteria 
- [ ] user can type in an email address and password 
- [ ] The system will find the email address in the database and verify the password submitted matches 
 
## Attachments 
 
- [Is child to System User Authentication](https://trello.com/c/XK8oNtaj/11-system-user-authentication#sf-dependencies&dependencyType=is_child_to&dependencyId=5f68d540d839a876b6bc7551&resolved=false) 
- [Is blocked by Setup the connection between Apollo Client and Lighthouse](https://trello.com/c/kYJzyych/67-setup-the-connection-between-apollo-client-and-lighthouse#sf-dependencies&dependencyType=is_blocked_by&dependencyId=5ff60dc216fe4c3e01d7d230&resolved=false) 
- [Is blocked by Setup Sanctum on the existing backend application](https://trello.com/c/ICQyqnbW/64-setup-sanctum-on-the-existing-backend-application#sf-dependencies&dependencyType=is_blocked_by&dependencyId=5ff5f3ba56787a54ba57b7af&resolved=false) 
- [Is blocked by Implement Apollo Client](https://trello.com/c/tYVRMBgk/58-implement-apollo-client#sf-dependencies&dependencyType=is_blocked_by&dependencyId=5ff5c69c1993136269e50656&resolved=false) 
 
> Trello Source: https://trello.com/c/qmezfF2g

**Comment by wreality** [comment]
@gmeben Is this a dupe of the work you're already doing in #32 and/or already covered by the scope of work defined in #32?

If yes, we should probably mark it as a dupe/close or delete it (my pref is to delete)

If no, we should probably add an estimate to it.

**Comment by gmeben** [comment]
@wreality This is the parent ticket of #32.

If you'd prefer to move away from the parent/child paradigm in our GitHub issues and close/delete this issue, I'm fine with that.

**Comment by gmeben** [comment]
@wreality Actually, upon further review, this is a sibling ticket to issue #32 and their shared parent is #84. 

In any case, I'm still fine with closing and deleting this issue. Its scope appears to overlap enough with #32 to consider them duplicates.
